### PR TITLE
Fix opam

### PR DESCRIPTION
**Pull request by eponier**
Several improvements to opam files

### CODE REVIEW

File: compiler/opam [hunk 1]
```diff
@@ -1,5 +1,6 @@
 opam-version: "2.0"
 name: "jasmin"
+version: "dev"
```
Comment (vbgl) [inline compiler/opam:3]: Can you please elaborate on why this change is important?

Should a similar patch be applied to the `opam` file for `mathcomp-word`?

Comment (eponier) [reply compiler/opam:3]: Actually, it's more that I wanted to apply this patch to `mathcomp-word` and decided to apply it here too. I think it's better practice in general that when you pin the git repo, you get "dev" as version name.

**Review by vbgl — APPROVED** [review]
